includes/discovery/stp.inc.php

```diff
@@ -126,12 +126,12 @@
             // set port binding
             $stp_port['port_id'] = dbFetchCell('SELECT port_id FROM `ports` WHERE `device_id` = ? AND `ifIndex` = ?', [$device['device_id'], $stp_raw[$port]['dot1dStpPort']]);
 
-            $dr = str_replace([' ', ':', '-'], '', strtolower($stp_raw[$port]['dot1dStpPortDesignatedRoot']));
-            $dr = substr($dr, -12); //remove first two octets
+            $dr = str_replace(['.', ' ', ':', '-'], '', strtolower($stp_raw[$port]['dot1dStpPortDesignatedRoot']));
```

Comment:
I would prefer LibreNMS\Util\Rewrite\macToHex()

Reply:
That's a point. Will check !

Reply:
Hi @mpikzink 
Not really possible cause there is no easy way to identify where the Mac Address starts. 
Basically, macToHex expects a mac address only, not a string ending with a mac address as we have here. And because we don't know where the mac address starts, we can only do the string replacements, and then keep the last 12 chars. 
So this PR cannot use macToHex